expression/util.go

```diff
@@ -30,16 +30,40 @@ import (
 
 // ExtractColumns extracts all columns from an expression.
 func ExtractColumns(expr Expression) (cols []*Column) {
+	result := make([]*Column, 0, 5)
+	return extractColumns(result, expr, nil)
+}
+
+// ExtractColumnsFromExpressions is a more effecient version of ExtractColumns for batch operation.
+// filter can be nil, or a function to filter the result column.
+// It's often observed that the pattern of the caller like this:
```

Comment:
Is there anywhere  in this repo matches this pattern?

Reply:
`server/util.go` for example.